packages/app/src-tauri/src/commands/transcription.rs

```diff
@@ -0,0 +1,212 @@
+    Ok(channel)
+}
+
+async fn whisper_transcriber_internal(
+    conf: WhisperConf,
+    mut producer: EventProducer,
+    receiver: CommandReceiver,
+) -> Result<()> {
+    let audio = load_audio(&conf.audio_path)?;
+    let audio_duration_secs = get_audio_duration_secs(&conf.audio_path).await?;
+    let (tx, mut rx) = mpsc::unbounded_channel();
+
+    let model = Whisper::builder()
+        .with_language(conf.language)
+        .with_source(conf.model)
+        .build()
+        .map_err(|_| Error::Whisper)?;
+
+    spawn(async move {
+        receiver.started().await;
+        producer.send(Ok(Event::Progress(0))).await;
+        let canc_token: CancellationToken = receiver.into();
+
+        if let Err(err) = model.transcribe_into(audio, tx).map_err(|_| Error::Whisper) {
+            producer.send(Err(err)).await;
+            canc_token.cancel();
+        }
+
+        loop {
+            select! {
+                _ = canc_token.cancelled() => {
+                    break
+                }
+                res = rx.recv() => {
+                    match res {
+                        Some(transcribed) => {
+                            let text = transcribed.text().to_owned();
+                            let probability_of_no_speech = transcribed.probability_of_no_speech();
+                            let offset = transcribed.start();
+                            let mut duration = transcribed.duration();
+
+                            //transcribed.duration() seems to be fixed at 30.
+                            //This can brake progress count when the last segment real duration <30
+                            if offset + duration > audio_duration_secs {
+                                duration = audio_duration_secs - offset;
+                            }
+
+                            producer.send(Ok(Event::TranscriptionSegment{text, probability_of_no_speech, offset, duration})).await;
+
+                            if audio_duration_secs >= 0.1 && audio_duration_secs.ceil() > duration.ceil() {
+                                let progress = (duration * 100. / audio_duration_secs).floor() as u8;
+                                producer.send(Ok(Event::DeltaProgress(progress))).await;
+                            }
+                        },
+                        None => {
+                            producer.send(Ok(Event::Progress(100))).await;
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+    });
+    Ok(())
+}
+
+fn load_audio(path: &Option<PathBuf>) -> Result<Decoder<BufReader<File>>> {
+    match path {
+        Some(path) => {
+            let file = BufReader::new(File::open(path)?);
+            Decoder::new(file).map_err(Into::into)
+        }
+        None => Err(Error::Decoder(DecoderError::UnrecognizedFormat)),
+    }
+}
+
+async fn get_audio_duration_secs(path: &Option<PathBuf>) -> Result<f64> {
```

Comment:
Can you expose a command for that, we use it in another part of the source (the react audio component).

Reply:
Added commands::ffmpeg::get_audio_duration_secs function